src/predicate_search_score_functions.py

```diff
@@ -129,16 +129,17 @@ def evaluate(self, candidate_predicates: FrozenSet[Predicate]) -> float:
         pruned_atom_data = utils.prune_ground_atom_dataset(
             self._atom_dataset,
             candidate_predicates | self._initial_predicates)
-        segments = [
-            seg for traj in pruned_atom_data
-            for seg in segment_trajectory(traj)
+        segmented_trajs = [
+            segment_trajectory(traj) for traj in pruned_atom_data
         ]
+        segments = [seg for traj in segmented_trajs for seg in traj]
         pnads = learn_strips_operators(segments, verbose=False)
         strips_ops = [pnad.op for pnad in pnads]
         option_specs = [pnad.option_spec for pnad in pnads]
         op_score = self.evaluate_with_operators(candidate_predicates,
-                                                pruned_atom_data, segments,
-                                                strips_ops, option_specs)
+                                                pruned_atom_data,
```

Comment:
can we get rid of `pruned_atom_data` everywhere now? (after `segmented_trajs` is created)